agent/consul/leader_connect_test.go

```diff
@@ -342,43 +340,27 @@ func TestCAManager_RenewIntermediate_Vault_Primary(t *testing.T) {
 				"Token":               testVault.RootToken,
 				"RootPKIPath":         "pki-root/",
 				"IntermediatePKIPath": "pki-intermediate/",
-				"LeafCertTTL":         "1s",
-				// The retry loop only retries for 7sec max and
-				// the ttl needs to be below so that it
-				// triggers definitely.
-				"IntermediateCertTTL": "5s",
+				"LeafCertTTL":         "2s",
+				"IntermediateCertTTL": "7s",
```

Comment:
This was another source of flakes. The leaf cert could expire before we got to checking if it was valid.

Increasing these values does not actually slow down the test at all , because we use a "NotBefore" date for the cert that is a full minute in the past, so we only end up waiting on the "renew interval", which is 200ms.